dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/organisationunit/OrganisationUnitProfileController.java

```diff
@@ -0,0 +1,96 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.organisationunit;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.IOException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.hisp.dhis.commons.jackson.config.JacksonObjectMapperConfig;
+import org.hisp.dhis.orgunitprofile.OrgUnitProfile;
+import org.hisp.dhis.orgunitprofile.OrgUnitProfileData;
+import org.hisp.dhis.orgunitprofile.OrgUnitProfileService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@Controller
+@RequestMapping( value = "/orgUnitProfile" )
+public class OrganisationUnitProfileController
+{
+    private final OrgUnitProfileService orgUnitProfileService;
+
+    private final ObjectMapper jsonMapper;
+
+    public OrganisationUnitProfileController( OrgUnitProfileService orgUnitProfileService, ObjectMapper jsonMapper )
+    {
+        checkNotNull( orgUnitProfileService );
+        checkNotNull( jsonMapper );
+
+        this.orgUnitProfileService = orgUnitProfileService;
+        this.jsonMapper = jsonMapper;
+    }
+
+    @PreAuthorize( "hasRole('ALL') or hasRole('F_ORG_UNIT_PROFILE_ADD')" )
+    @PostMapping( consumes = "application/json" )
+    @ResponseStatus( HttpStatus.NO_CONTENT )
+    public void saveProfile( HttpServletRequest request, HttpServletResponse response )
+        throws IOException
+    {
+        OrgUnitProfile profile = jsonMapper.readValue( request.getInputStream(), OrgUnitProfile.class );
+
+        orgUnitProfileService.saveOrgUnitProfile( profile );
+    }
+
+    @GetMapping( produces = MediaType.APPLICATION_JSON_VALUE )
+    public void getProfile( HttpServletResponse response )
+        throws IOException
+    {
+        jsonMapper.writeValue( response.getOutputStream(), orgUnitProfileService.getOrgUnitProfile() );
```

Comment:
With RestController, could we just return the profile object? Might be a reason I don't see.

Reply:
fixed